contrib/submit-queue/submit-queue.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+// A simple binary for merging PR that match a criteria
+// Usage:
+//   submit-queue -token=<github-access-token> [-min-pr-number=<number>] [-dry-run] [-once]
+//
+// Details:
+// Usage of ./submit-queue:
+//  -alsologtostderr=false: log to standard error as well as files
+//  -dry-run=false: If true, don't actually merge anything
+//  -log_backtrace_at=:0: when logging hits line file:N, emit a stack trace
+//  -log_dir="": If non-empty, write log files in this directory
+//  -logtostderr=false: log to standard error instead of files
+//  -min-pr-number=0: The minimum PR to start with [default: 0]
+//  -one-off=false: If true, only merge one PR, don't run forever
+//  -stderrthreshold=0: logs at or above this threshold go to stderr
+//  -token="": The OAuth Token to use for requests.
+//  -v=0: log level for V logs
+//  -vmodule=: comma-separated list of pattern=N settings for file-filtered logging
+
+import (
+	"errors"
+	"flag"
+
+	"github.com/GoogleCloudPlatform/kubernetes/contrib/submit-queue/github"
+	"github.com/GoogleCloudPlatform/kubernetes/contrib/submit-queue/jenkins"
+
+	"github.com/golang/glog"
+	github_api "github.com/google/go-github/github"
+)
+
+var (
+	token       = flag.String("token", "", "The OAuth Token to use for requests.")
+	minPRNumber = flag.Int("min-pr-number", 0, "The minimum PR to start with [default: 0]")
+	dryrun      = flag.Bool("dry-run", false, "If true, don't actually merge anything")
+	oneOff      = flag.Bool("once", false, "If true, only merge one PR, don't run forever")
+	job         = flag.String("jenkins-job", "kubernetes-e2e-gke-ci", "The job in Jenkins to use for stability testing")
+)
+
+const (
+	user    = "GoogleCloudPlatform"
```

Comment:
user isn't a great name, since it could also potentially refer to the user doing the merges